File: week6/ee/수들의_합_2.js
const fs = require("fs")
const filePath = process.platform === "linux" ? "/dev/stdin" : "./2003.txt"

const input = fs.readFileSync(filePath).toString().trim().split("\n")

const [N, M] = input[0].split(" ").map(Number)
const arr = input[1].split(" ").map(Number)

let cnt = 0
let cur_sum = 0

let end = 0
for (let start = 0; start < N; start++) {
  while (cur_sum < M && end < N) {
    cur_sum += arr[end]
    end++
  }

  if (cur_sum == M) {
    cnt++
  }

  cur_sum -= arr[start]
}

console.log(cnt)

File: week06/ee/수들의_합_2.py
import sys
sys.stdin = open("2003.txt")

N, M = map(int, input().split())
arr = list(map(int, input().split()))

cnt = 0
cur_sum = 0

end = 0
for start in range(N):
    while cur_sum < M and end < N:
        cur_sum += arr[end]
        end += 1
    
    if cur_sum == M:
        cnt += 1
    
    cur_sum -= arr[start]

print(cnt)